genesyscloud/resource_genesyscloud_widget_deployment.go

```diff
@@ -0,0 +1,405 @@
+package genesyscloud
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/url"
+	"time"
+
+	"github.com/hashicorp/go-cty/cty"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/mypurecloud/platform-client-sdk-go/v56/platformclientv2"
+)
+
+const (
+	V1            = "v1"
+	V1HTTP        = "v1-http"
+	V2            = "v2"
+	THIRDPARTY    = "third-party"
+	HTTPSPROTOCOL = "https"
+	WEBSKINBASIC  = "basic"
+	WEBSKINMODERN = "modern-caret-skin"
+)
+
+var (
+	clientConfigSchemaResource = &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"webchat_skin": {
+				Description:  "Skin for the webchat user. (basic, modern-caret-skin)",
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{WEBSKINBASIC, WEBSKINMODERN}, false),
+			},
+			"authentication_url": {
+				Description:      "Url endpoint to perform_authentication",
+				Type:             schema.TypeString,
+				Required:         false,
+				Optional:         true,
+				ValidateDiagFunc: validateAuthURL,
+			},
+		},
+	}
+)
+
+func getAllWidgetDeployments(_ context.Context, clientConfig *platformclientv2.Configuration) (ResourceIDMetaMap, diag.Diagnostics) {
+	resources := make(ResourceIDMetaMap)
+	widgetsAPI := platformclientv2.NewWidgetsApiWithConfig(clientConfig)
+
+	for pageNum := 1; ; pageNum++ {
```

Comment:
Looks like there's no need for a for loop here because the request can't be paginated

Reply:
Doh!  Took it out.  That's what I get for coding via cut and paste